templates/net_main/commands/build.sh.j2

```diff
@@ -0,0 +1,31 @@
+{
+set -e
+
+{% set template = 'ethernet' if vlan is None else 'vlan' %}
```

Comment:
I asked the question to Jerry last week about different primitives each one for ethernet and vlan, but he said that same primitive with IF condition can be used as the procedure to setup interface as system service is same, ie creating `up`, `down` and `service` file.

As I wrote now, to setup vlan tagged interface, this primitive needs to be called twice, But I realized this is the case of `Interference`. Will discuss this in next meeting I think this is a good example to deal with interference.

Reply:
If it was the case that it was one template with `if vlan is not None: do something` then it would not have raised any concerns to me. It would be the same as `if len(routes) > 0: do something`. 

As it is written now, it is different templates and different service files which is why I raised my concerns.

Reply:
I was just trying to keep less `if` conditions or try to avoid them in the build.sh.j2 by separating them as different files. Do you want me to put them in the same template with of course one `if` condition required?

Reply:
My interpretation of this is having one template to create the ethernet and vlan interface service up and down files removes the interference you referred to and ability to create in one primitive request. 

To me it is confusing as to why the same primitive is called twice to create the service files for a vlan tagged interface. I see `vlans` as an optional stanza  similar to `routes` and `ip_addresses` for the interface.

Reply:
Yes, `vlans` as an optional similar to `routes` would avoid to call the primitive twice.